### ISSUE TITLE
Use of angle brackets around file names for include statements

### ISSUE THREAD

**Issue by elfring** [FidoProject/Fido#51]
Would you like to [replace any double quotes](https://github.com/FidoProject/Fido/blob/0af4306319f6d9d8243c20babddb3c565277791a/include/Fido.h#L4 "Check header file inclusion style") by [angle brackets around file names for include statements](http://stackoverflow.com/questions/21593/what-is-the-difference-between-include-filename-and-include-filename "What is the difference between #include <filename> and #include “filename”?")?

**Comment by truell20** [comment]
@elfring Doesn't that mean that the library would have to be globally installed for it to build and run properly? Or would the `include` folder have to be marked as an include path by g++? Either way, what is the advantage over the double quotes?

**Comment by elfring** [comment]
I suggest to reconsider the consequences of [the following wording](http://c0x.coding-guidelines.com/6.10.2.html "Source file inclusion") from [the section "16.2 Source file inclusion"](http://open-std.org/jtc1/sc22/wg21/docs/papers/2013/n3797.pdf "C++ working draft") in the standard specification for the programming language "C++".
```
…
The named source file is searched for in an implementation-defined manner. If this search is not supported, or if the search fails, the directive is reprocessed as if it read

#include <h-char-sequence> new-line
…
```

* How do you think about to avoid "a duplicated file search" here?
* Would you like to [restrict the searched directories](https://gcc.gnu.org/onlinedocs/cpp/Include-Syntax.html "Include syntax description") for header files of your software?

**Comment by truell20** [comment]
@FlyingGraysons @joshuagruenstein and I are a little unsure of what you are saying. Could you please clarify, @elfring?

**Comment by elfring** [comment]
There are different opinions about the handling of the involved implementation-defined behaviour.
* Will header files be also searched outside the specified include directories if double quotes are used for the discussed preprocessor statement?
* Is there a speed difference measurable if a file is not found there and the search will be retried with "the angle brackets inclusion method"?

**Comment by truell20** [comment]
I believe that:
* If double quotes are used, only the local and include directories are searched
* If brackets are used, the system and include directories are search are search

I doubt that there is a measurable speed difference between the two methods.

I may be wrong though.

**Comment by elfring** [comment]
Can you imagine use cases where you would like to exclude the search in "local" directories?

**Comment by truell20** [comment]
I cannot

**Comment by truell20** [comment]
@elfring Can you think of any use cases?

**Comment by elfring** [comment]
Yes. - This is the reason for my feature request here.

How much do you care where the include files are safely stored for a software library?

**Comment by FlyingGraysons** [comment]
@elfring What is the case?

**Comment by elfring** [comment]
I find it important to be reasonably sure that the desired computations were really correct because it could be determined by the applied software build parameters that the used functions were actually derived from the expected source files. Can you follow this design concern?

**Comment by truell20** [comment]
What do you mean by this: "the desired computations were really correct"?

**Comment by elfring** [comment]
I would generally prefer to exclude the potential for the inclusion of header files from unexpected directories. Can such a security detail become a bit more important occasionally?

**Comment by truell20** [comment]
Yes, but the brackets mean that a person would have to specify an include directory, which is slightly annoying.

**Comment by elfring** [comment]
Will the mentioned "annoyance" vanish after safe include specifications would be used?

**Comment by truell20** [comment]
Yes. This does sound like a good idea. Thank you for the feature request.

**Comment by truell20** [comment]
Would specifying the include directory in our g++ compilation command in our makefiles and switching the the brackets fix your specified problem?

**Comment by elfring** [comment]
How would you like to distinguish library versions by your software build system?

**Comment by truell20** [comment]
Not sure? Any suggestions?

**Comment by truell20** [comment]
Feel free to submit a pull request that fixes the problem @elfring
